sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java

```diff
@@ -107,4 +113,19 @@ private boolean alreadyAuthenticated() {
                        .filter(r -> !r.isEmpty())
                        .orElse(DEFAULT_AUTHORITY_SET);
     }
+
+    protected Set<SimpleGrantedAuthority> scopeToSimpleGrantedAuthoritySet(UserPrincipal userPrincipal) {
+        Set<SimpleGrantedAuthority> scopeSimpleGrantedAuthoritySet =
+            Optional.of(((String)userPrincipal.getClaim(AADTokenClaim.SCP)))
+                .map(scopes->scopes.split(" "))
+                .map(Arrays::stream)
+                .orElseGet(Stream::empty)
+                .filter(StringUtils::hasText)
+                .map(s -> new SimpleGrantedAuthority(SCOPE_PREFIX + s))
+                .collect(Collectors.toSet());
+
+        return Optional.of(scopeSimpleGrantedAuthoritySet)
+            .filter(r -> !r.isEmpty())
+            .orElse(DEFAULT_AUTHORITY_SET);
```

Comment:
We don't need this here, we are parsing scopes in this function, it's not reasonable to output a set of ROLE_xxx.

Reply:
yes,I'm going to modify it